pkg/tenant/members.go

```diff
@@ -361,6 +361,41 @@ func (s *Server) MemberRoleUpdate(c *gin.Context) {
 		return
 	}
 
+	// Check to ensure the memberID from the URL matches the member ID from the database.
+	if memberID.Compare(member.ID) != 0 {
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("member id does not match id in URL"))
+		return
+	}
+
+	// If the member to be updated is an owner, loop over dbMember and break out of the loop if there are at least two owners.
+	// If member is the only owner, their role cannot be changed.
+	if member.Role == perms.RoleOwner {
+		// Get members from the database and set page size to return all members.
+		// TODO: Create list method that will not require pagination for this endpoint.
+		getAll := &pg.Cursor{StartIndex: "", EndIndex: "", PageSize: 100}
+		var members []*db.Member
+		if members, _, err = db.ListMembers(c, orgID, getAll); err != nil {
+			sentry.Error(c).Err(err).Msg("could not list members")
+			c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not get member from the database"))
+			return
+		}
+
+		count := uint8(0)
```

Comment:
Any particular reason why this is a `uint8`? Since it's a temporary variable I think it's okay to be a normal, signed int. If we were storing it, we might consider making it unsigned.

Reply:
I was thinking of being more specific about the type of number that should be returned here, but this can be an `int`.

Reply:
Alright, I don't have too strong of an opinion on it, but keep in mind that with `uint8` it will have a max value of 255, maybe just to be safe we should use `uint16`.

Also since the count is being returned from the method we could just do:

`var count uint8`

instead of manually initializing it to 0.

Reply:
The loop should break at 2 owners, so that's why I went with uint8. It's not a big thing, so I'll use `int`.